security/pfSense-pkg-zeek/files/usr/local/pkg/zeek.inc

```diff
@@ -0,0 +1,982 @@
+*/
+
+require_once('config.inc');
+require_once('globals.inc');
+require_once('notices.inc');
+require_once('pfsense-utils.inc');
+require_once('pkg-utils.inc');
+require_once('services.inc');
+require_once('service-utils.inc');
+require_once("functions.inc");
+require_once('util.inc');
+require_once("interfaces.inc");
+require_once('xmlrpc_client.inc');
+
+if (!function_exists("filter_configure")) {
+	require_once("filter.inc");
+}
+
+$shortcut_section = "zeek";
+
+define('ZEEK_LOCALBASE', '/usr/local');
+define('ZEEK_ETC', ZEEK_LOCALBASE .'/etc');
+define('ZEEK_LOGDIR', ZEEK_LOCALBASE .'/logs/current');
+define('ZEEK_SITEDIR', ZEEK_LOCALBASE .'/share/zeek/site');
+
+/*
+* Utility functions
+*/
+
+/* Handle base64 encoding and linebreaks in textarea configuration fields*/
+function zeek_text_area_decode($text) {
+	return preg_replace('/\r\n/', "\n", base64_decode($text));
+}
+
+/* Get interface IP and netmask for Zeek interfaces*/
+function zeek_get_real_interface_address($iface) {
+	if (!function_exists("get_interface_ip")) {
+		require_once("interfaces.inc");
+	}
+	return array(get_interface_ip($iface), gen_subnet_mask(get_interface_subnet($iface)));
+}
+
+/* Check whether Zeek is enabled*/
+function zeek_enabled() {
+	global $config, $zeek_enabled;
+	$zeek_enabled = false;
+
+	if (is_array($config['installedpackages']['zeek']['config'])) {
+		// check whether Zeek is enabled ...
+		if ($config['installedpackages']['zeek']['config'][0]['enable_zeek'] == "on") {
+			// ... and has at least one interface configured ...
+			if (!empty($config['installedpackages']['zeek']['config'][0]['active_interface'])) {
+				$zeek_enabled = true;
+			}
+		}
+	}
+	return $zeek_enabled;
+}
+
+// Check whether Zeek Cluster is enabled
+function zeek_cluster_enabled() {
+	global $config, $zeek_cluster_enabled;
+	$zeek_cluster_enabled = false;
+
+	if (is_array($config['installedpackages']['zeekcluster'])) {
+		// check whether zeek cluster is enabled ...
+		if ($config['installedpackages']['zeekcluster']['config'][0]['enable_cluster'] == "on"){
+			$zeek_cluster_enabled = true;
+		}
+	}
+	return $zeek_cluster_enabled;
+}
+
+// Start and/or stop services according to Zeek configuration
+function zeek_restart_services() {
+	global $config;
+	// do not (re)start zeek services on boot
+	if (platform_booting()) {
+		return;
+	}
+
+	if (zeek_enabled()) {
+		if (!is_service_running('zeek')) {
+			log_error("[zeek] Starting service...");
+			mwexec(ZEEK_ETC . "/rc.d/zeek.sh start");
+		} else {
+			log_error("[zeek] Reloading for configuration sync...");
+			mwexec(ZEEK_ETC . "/rc.d/zeek.sh restart");
+		}
+		// sleep for a couple seconds to give zeek a chance to fire up fully.
+		for ($i = 0; $i < 10; $i++) {
+			if (!is_service_running('zeek')) {
+				sleep(1);
+			}
+		}
+	} else {
+		//Zeek is disabled
+		if (is_service_running('zeek')) {
+			log_error("[zeek] Stopping service...");
+			//stop_service("zeek");
+			mwexec(ZEEK_ETC . "/rc.d/zeek.sh stop");
+		}
+	}}
```

Comment:
Extra curly brace here should probably be on the next line with no indent (if it's the end of the function, which it appears to be)